### DISCUSSION TITLE
How not to close dialog when clicking outside of popup menu, that opens in that dialog?

### DISCUSSION THREAD

**Discussion by FragsterAt** [quasarframework/quasar#16527] [General - Components / Directives / etc]
https://codepen.io/FragsterAt/pen/mdvVbQM?editors=101
Click on button, then click in button in dialog, then click outside of menu and dialog. Both will be closed.
I need to close menu, but not close the dialog. But dialog must stay not persistent (user should be able to close it with a second click outside)

Also in real app popup menu is part of complex component, so i can't control 'persistent' prop of dialog from it.

May be there is any combinations of props, that could help me?

**Comment by metalsadman** [comment]
https://codepen.io/metalsadman/pen/mdvVGba?editors=101

**Reply by FragsterAt** [reply]
You just add 'permanent' prop to dialog when menu is shown. But in real life it is component inside component inside component, so this method is not suitable.

**Reply by metalsadman** [reply]
use a store for long-range props.

**Reply by FragsterAt** [reply]
This is a universal ipnut component with q-date in the pop-up menu. There may be many instances of this component in different places of the application, I can't imagine how store can be used here

**Reply by metalsadman** [reply]
try to make a close reproduction of what you have, maybe in a stackblitz.
